aemcmc/utils.py

```diff
@@ -0,0 +1,64 @@
+from dataclasses import dataclass, field
+from typing import Dict, Optional, Tuple
+
+from aesara.graph.basic import Variable
+from aesara.tensor.var import TensorVariable
+
+
+@dataclass(frozen=True)
+class ModelInfo:
+    """A class that tracks sampler-specific variable types, names, and updates for a model graph."""
+
+    observed_rvs: Tuple[TensorVariable, ...]
+    """Observed random/measurable variables."""
+    rvs_to_values: Dict[TensorVariable, TensorVariable]
+    """A map between random/measurable variables and their value variables."""
+    deterministic_vars: Tuple[TensorVariable, ...]
+    """Stochastic variables that are tracked but not sampled directly/explicitly."""
+    updates: Optional[Dict[Variable, TensorVariable]]
+    """Updates to be passed to `aesara.function`."""
```

Comment:
This confused me when I first read the code. Maybe it would be good to be slightly more specific and add ", for instance when the model graph contains a `scan` loop."

Reply:
Yeah, I'll add that.